### PR TITLE
executor: parallel some part of the sampling-based analyze

### PR DESCRIPTION
**Pull request by winoros**
### What problem does this PR solve?

Problem Summary:

Improve the speed of the new analyze tech.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

Parallel the part of merge the regions' data and building the stats for columns and indexes.

### Check List 

Tests 

- Existing Unit test(All existing tests passed)
- Integration test(Tested all large datasets with successful and correct analyze result).

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM

### Release note 

- No release note

### CODE REVIEW

File: executor/analyze.go [hunk 1]
```diff
@@ -730,69 +752,203 @@ func (e *AnalyzeColumnsExec) buildSamplingStats(ranges []*ranger.Range) (
...
+		subCollector.FromProto(colResp.RowCollector)
+		e.job.Update(subCollector.Count)
+		retCollector.MergeCollector(subCollector)
+	}
+	resultCh <- &samplingMergeResult{collector: retCollector}
+}
+
+type samplingBuildTask struct {
+	id               int64
+	collectorBuilder func() (*statistics.SampleCollector, error)
```
Comment (time-and-fate) [inline executor/analyze.go:864]: Seems this is not used.

File: executor/analyze.go [hunk 2]
```diff
@@ -730,69 +752,203 @@ func (e *AnalyzeColumnsExec) buildSamplingStats(ranges []*ranger.Range) (
...
 	}
-	hists = make([]*statistics.Histogram, 0, len(e.colsInfo))
-	topns = make([]*statistics.TopN, 0, len(e.colsInfo))
-	fmSketches = make([]*statistics.FMSketch, 0, len(e.colsInfo))
+	totalLen := len(e.colsInfo) + len(e.indexes)
+	hists = make([]*statistics.Histogram, totalLen)
+	topns = make([]*statistics.TopN, totalLen)
+	fmSketches = make([]*statistics.FMSketch, 0, totalLen)
+	buildResultChan := make(chan error, len(e.colsInfo)+len(e.indexes))
+	buildTaskChan := make(chan *samplingBuildTask, len(e.colsInfo)+len(e.indexes))
```
Comment (time-and-fate) [inline executor/analyze.go:760]: Let's use `totalLen` directly. :joy:

**Review by qw4990 — APPROVED** [review]

**Pull request merged**